astroplan/core.py

```diff
@@ -649,10 +734,20 @@ def calc_rise(self, time, target, which='nearest', horizon=0*u.degree):
                 return previous_rise
 
         if which == 'nearest':
-            if abs(time - previous_rise) < abs(time - next_rise):
-                return previous_rise
+            if _target_is_vector(target):
```

Comment:
You're right. Originally I was under the impression that the `Time` initializer would keep a nan without issue, but when you get a `ValueError` when you do. This leads to the question of how best to handle bad times. I think I like the list of `Time`'s with nan's where appropriate sounds OK.